src/internal/_assocPath.js

```diff
@@ -0,0 +1,11 @@
+var _assoc = require('./_assoc');
+var _slice = require('./_slice');
+
+
+module.exports = function _assocPath(path, val, obj) {
```

Comment:
It's a question of philosophy. One can make an argument for either approach. One could argue that `_assocPath` (and by extension `assocPath`) should *always* return an object to which no one holds a reference. On the other hand one could argue that since Ramda promotes programming without mutation, cloning is unnecessary.